src/Symfony/Component/HttpKernel/DependencyInjection/AddCacheWarmerPass.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+
+/**
+ * Registers the cache warmers.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class AddCacheWarmerPass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    private $tag;
+    private $definition;
+
+    public function __construct($definition = null, $tag = null)
```

Comment:
What about making them required now, and call `parent::__construct` from the legacy class?